conanfile.py

```diff
@@ -101,6 +102,9 @@ def build_requirements(self):
         if self.options.build_docs:
             self.build_requires("doxygen/1.9.4") # doxygen/1.9.5 will update dependency on zlib/1.2.12 to zlib/1.2.13
 
+        if self.settings.compiler in ["gcc", "clang"]:
+            self.build_requires("mold/1.11.0")      
```

Comment:
```suggestion
            self.build_requires("mold/1.11.0") 
            self.build_requires("openssl/3.1.2", override=True)
```

Here is where we should override openssl only when using the mold linker, but only as a build requirement as its not directly used by the build (note I don't think this make much difference to how Conan internally represent it other than it will contain a `build` flag: .https://github.com/conan-io/conan/blob/f9b6fc99677e3c7d54db5ebbff29d5af72561a0b/conans/model/requires.py#L500  

I've hit problems with OpenSSL before requiring overriding this (I think when the CMake package has a different version to used elsewhere) so it not uncommon to need to override it like this.

Reply:
Clear, thanks for the explanation.